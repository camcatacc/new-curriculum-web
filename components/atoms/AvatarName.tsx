import { Typography } from "@material-ui/core";
import React from "react";
import { InlineStyles } from "../../styles/stylesInterface";

export const AvatarName = (props: { name: string; surname: string }) => {
    return (
        <Typography variant="h3" align="center">
            {props.name} <br />
            <Typography
                variant="inherit"
                className="leading-loose"
                style={{ fontSize: "smaller" }}
            >
                {props.surname}
            </Typography>
        </Typography>
    );
};
