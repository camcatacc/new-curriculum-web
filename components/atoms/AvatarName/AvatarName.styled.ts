// Modules
import styled from "styled-components";

// Elements
import { Typography as MaterialTypography } from "@material-ui/core";

// Components
export const Typography = styled(MaterialTypography)<{ $color: string }>`
	color: ${({ $color }) => $color};
`;
