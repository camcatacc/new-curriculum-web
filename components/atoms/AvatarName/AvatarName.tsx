// Modules
import React from "react";

// Elements
import { Typography } from "@material-ui/core";

// Definitions
export interface AvatarNameProps {
	name: string;
	surname: string;
}

// Element
const AvatarName = ({ name, surname }: AvatarNameProps) => {
	return (
		<Typography variant="h3" align="center">
			{name} <br />
			<Typography variant="inherit" className="leading-loose" style={{ fontSize: "smaller" }}>
				{surname}
			</Typography>
		</Typography>
	);
};

export default AvatarName;
