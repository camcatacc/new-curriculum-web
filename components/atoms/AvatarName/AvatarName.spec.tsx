// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import AvatarName from "components/atoms/AvatarName/AvatarName";

// Tests
describe("AvatarName", () => {
	it("Renders as expected", () => {
		const { container } = render(<AvatarName name="Marco" surname="Polo" />);
		expect(container).toMatchSnapshot();
	});
});
