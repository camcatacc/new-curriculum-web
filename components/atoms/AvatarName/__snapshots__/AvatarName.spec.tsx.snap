// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarName Renders as expected 1`] = `
<div>
  <h3
    class="MuiTypography-root sc-bdvvtL iytgcZ MuiTypography-h3 MuiTypography-alignCenter"
  >
    Marco
     
    Polo
  </h3>
</div>
`;
